argus/unit_tests/action_manager/test_windows.py

```diff
@@ -13,64 +13,818 @@
+            with self.assertRaises(download_exc):
+                self._action_manager.sysprep()
+            self._client.run_remote_cmd.assert_not_called()
+            return
+
+        if wait_exc:
+            mock_wait_boot_completion.side_effect = wait_exc
+            with self.assertRaises(wait_exc):
+                self._action_manager.sysprep()
+            return
+
+        if run_exc:
+            self._client.run_remote_cmd.side_effect = run_exc
+            with test_utils.LogSnatcher('argus.action_manager.windows.Windows'
+                                        'ActionManager.sysprep') as snatcher:
+                self.assertIsNone(self._action_manager.sysprep())
+            self.assertEqual(snatcher.output[-2:],
+                             ['Currently rebooting...',
+                              'Wait for the machine to finish rebooting ...'])
+        else:
+            self.assertIsNone(self._action_manager.sysprep())
+
+        mock_download_resource.assert_called_once_with(
+            test_utils.SYSPREP_RESOURCE_LOCATION, cmd)
+        mock_wait_boot_completion.assert_called_once_with()
+
+    def test_sysprep_successful(self):
+        self._test_sysprep()
+
+    def test_sysprep_download_resource_fail(self):
+        self._test_sysprep(download_exc=exceptions.ArgusTimeoutError)
+
+    def test_sysprep_wait_boot_completion_fail(self):
+        self._test_sysprep(wait_exc=exceptions.ArgusTimeoutError)
+
+    def test_sysprep_run_remote_cmd_exc(self):
+        self._test_sysprep(run_exc=requests.Timeout)
+
+    @mock.patch('argus.action_manager.windows.WindowsActionManager'
+                '.exists')
     def test_git_clone_exists(self, mock_exists):
         mock_exists.return_value = True
-        self.assertRaises(exceptions.ArgusCLIError,
-                          self.action_manager.git_clone,
-                          'fake_url', 'fake_location')
 
-    @mock.patch('argus.action_manager.windows.WindowsActionManager.'
-                'exists')
+        with self.assertRaises(exceptions.ArgusCLIError):
+            self._action_manager.git_clone(test_utils.URL, test_utils.LOCATION)
+
+    @mock.patch('argus.action_manager.windows.WindowsActionManager'
+                '.exists')
     def test_git_clone_could_not_clone(self, mock_exists):
         mock_exists.return_value = False
-        res = self.action_manager.git_clone('fake_url', 'location', count=0)
-        self.assertFalse(res)
 
-    @mock.patch('argus.action_manager.windows.WindowsActionManager.'
-                'exists')
+        self.assertFalse(self._action_manager.git_clone(
+            test_utils.URL, test_utils.LOCATION, count=0))
+
+    @mock.patch('argus.action_manager.windows.WindowsActionManager'
+                '.exists')
     def test_git_clone_successful(self, mock_exists):
         mock_exists.return_value = False
-        self.action_manager._client.run_command.return_value = True
```

Comment:
You mocked `run_command` to return `True` value but the real `run_command` method will never return `True`.